chain/chaindb_test.go

```diff
@@ -0,0 +1,97 @@
+package chain
+
+import (
+	"github.com/aergoio/aergo-lib/db"
+	"github.com/aergoio/aergo/v2/consensus"
+	"github.com/stretchr/testify/assert"
+	"sync/atomic"
+	"testing"
+)
+
+func TestChainDB_hardforkHeights(t *testing.T) {
+	type fields struct {
+		cc        consensus.ChainConsensus
+		latest    atomic.Value
+		bestBlock atomic.Value
+		store     db.DB
+	}
+	tests := []struct {
+		name   string
+		hardfork string
+		wantSuccess bool
+	}{
+		{ "normal", `{"V2":2000,"V3":30000,"V4":400000}`, true},
+		{ "wrong", `{"A":2000,"B3A":30000,"V4":400000}`, false},
+		{ "wrong2", `{"A2":2000,"B3":30000,"V4":400000}`, false},
```

Comment:
here is one place

Reply:
This is simulate the chaindb, not rpc response. 
The chaindb stores hardfork status with json string since hardfork V2. We cannot change the internal data structures without migration.